claudia-lab/server.js

```diff
@@ -0,0 +1,66 @@
+'use strict';
+
+//node modules
+const http = require('http');
+const url = require('url');
+const queryString = require('querystring');
+
+// npm modules
+const cowsay = require('cowsay');
+
+// app moduels
+const parseBody = require('./lib/parsebody.js');
+
+// module constats
+const PORT = process.env.PORT || 3000;
+
+//creates server and takes a callback that handles the request and sends a response
+const server = http.createServer(function(req, res) {
+  req.url = url.parse(req.url); //gets url
+  req.url.query = queryString.parse(req.url.query); //gets query string
+
+  if (req.url.pathname === '/') {
+    res.writeHead(200, {'Content-Type': 'text/plain'});
+    res.write('Hello World');
+    res.end();
+    return;
+  }
+
+  if (req.method === 'GET' && req.url.pathname === '/cowsay') {
+    if (req.url.query.text) {
+      res.writeHead(200, {'Content-Type': 'text/plain'});
+      res.write(cowsay.say({text:req.url.query.text, f:req.url.query.f}));
+      res.end();
+    } else {
+      res.writeHead(400, {'Content-Type': 'text/plain'});
+      res.write(cowsay.say({text: 'bad request\ntry: localhost:3000/cowsay?text=howdy'}));
+      res.end();
+    }
+    return;
+  }
+
+  if (req.method === 'POST' && req.url.pathname === '/cowsay'){
+    parseBody(req, function(err){
+      if (err) return console.error(err);
+
+      if (req.body.text) {
+        res.writeHead(200, {'Content-Type': 'text/plain'});
+        res.write(cowsay.say({text:req.body.text, f:req.url.body.text}));
```

Comment:
@loomnugget the `f` as the property name is a bit unclear - this should be renamed to an actual word to help with overall readability.  I don't want to move you away from **ever** using a single letter variable, as this is often utilized when working with implicit programming styles, like the map, filter, and reduce methods - for example, if we want to represent a number for each item in an array, `n` might be an appropriate option as other developers will surely know what you mean:

```
var nums = [1, 2, 3, 4, 5]

nums.map(function(n) {
  return n + n;
});

>> [2, 4, 6, 8, 10]
```